uiowa/uiowa
# [Icons] ui_icons integration

Resolves https://github.com/uiowa/uiowa/issues/8212
Resolves https://github.com/uiowa/uiowa/issues/6978
Resolves https://github.com/uiowa/uiowa/issues/4290

# How to test

```
ddev composer install && ddev blt frontend  && ddev blt ds --site=sandbox.uiowa.edu && ddev drush @sandbox.local uli /node/21/layout
```

1. Review https://github.com/uiowa/uids/pull/948
2. Code review
3. Edit a card and add an icon, confirm that only the two-color icons are available and the image should be unset on block save
4. Page through the entire library of the icon picker and make sure none of the thumbnails are broken or appear off-centered
5. Edit the same card and add an image, the icon should be unset on block save
6. Confirm that image or icon is saved and displayed in the radios after saving a block
7. Test adding icons to different card variants
8. ~See if you can confirm the bug on https://www.drupal.org/project/ui_icons/issues/3491714.~
     1. ~Download https://gitlab.com/ui-icons/ui-icons-example/-/tree/main/ui_icons_drupal and put it in modules/custom and enable it on the sandbox site and then follow the instructions in the drupal.org issue.~

## Questions
**How to remove the unused brand_icon media entity?** 
This pr removes the brand_icon media entity.   Syncing a site without https://github.com/uiowa/uiowa/pull/8298/commits/6bc1f1bb95226a5f68ec3439bd20fe8cb59cd756 caused the snyc to fail.  Should we delete these files once the brand_icon media entity has been deleted in a future pull request?


**What happens if an icon is removed from a pack?**
After a `cr` the field value gets removed and it doesn't attempt to display an image.

**~Can multiple icon packs be used while limiting the icon field to only use the brand icon set?~**
~It appears there is a bug for that on the field.  If we installed the brand icon set and the font awesome set, both are appearing available in the icon picker for the card.  I've added a bug report to https://www.drupal.org/project/ui_icons/issues/3491714.~ Fixed.  

## How to add new icons in the future

-  Attach new hash from https://github.com/uiowa/brand-icons to uids_base package.json
